cash/cash.d.ts

```diff
@@ -0,0 +1,495 @@
+
+  /**
+   * parent
+   * Returns parent element.
+   */
+  parent(): Cash;
+
+  /**
+   * parents
+   * Returns collection of elements who are parents of element. Optionally filtering by selector.
+   */
+  parents(selector?: string): Cash;
+
+  /**
+   * position
+   * Get the coordinates of the first element in a collection relative to its offsetParent.
+   */
+  position(): OffsetType;
+
+  /**
+   * prepend
+   * Prepends element to the each element in collection.
+   */
+  prepend(content: string): Cash;
+
+  /**
+   * prependTo
+   * Prepends elements in a collection to the target element(s).
+   */
+  prependTo(parent: string|Element|Cash): Cash;
+
+  /**
+   * prev
+   * Returns the previous adjacent element.
+   */
+  prev(): Cash;
+
+  /**
+   * prop
+   * Returns a property value when just property is supplied.
+   * Sets a property when property and value are supplied, and sets multiple properties when an object is supplied.
+   */
+  prop(name: string): any;
+  prop(name: string, value: string): Cash;
+
+  /**
+   * ready
+   * Calls callback method on DOMContentLoaded.
+   */
+  ready(fn: Function): void;
+
+  /**
+   * remove
+   * Removes collection elements from the DOM.
+   */
+  remove(): Cash;
+  
+  /**
+   * removeAttr
+   * Removes attribute from collection elements.
+   */
+  removeAttr(name: string): Cash;
+
+  /**
+   * removeClass
+   * Removes className from collection elements.
+   * Accepts space-separated classNames for removing multiple classes.
+   * Providing no arguments will remove all classes.
+   */
+  removeClass(c?: string): Cash;
+
+  /**
+   * removeData
+   * Removes linked data and data-attributes from collection elements.
+   */
+  removeData(key: string): Cash;
+
+  /**
+   * removeProp
+   * Removes property from collection elements.
+   */
+  removeProp(name: string): Cash;
+
+  /**
+   * serialize
+   * When called on a form, serializes and returns form data.
+   */
+  serialize(): string;
+
+  /**
+   * siblings
+   * Returns a collection of sibling elements.
+   */
+  siblings(): Cash;
+
+  /**
+   * text
+   * Returns the inner text of the first element in the collection, sets the text if textContent is provided.
+   */
+  text(): string;
+  text(content?: string): Cash;
+
+  /**
+   * toggleClass
+   * Adds or removes className from collection elements based on if the element already has the class.
+   * Accepts space-separated classNames for toggling multiple classes, and an optional force boolean to ensure classes are added (true) or removed (false).
+   */
+  toggleClass(c: string, state?: boolean): Cash;
+
+  /**
+   * trigger
+   * Triggers supplied event on elements in collection. Data can be passed along as the second parameter.
+   */
+  trigger(eventName: string, data?: any): Cash;
+
+  /**
+   * val
+   * Returns an inputs value. If value is supplied, sets all inputs in collection's value to the value argument.
+   */
+  val(): any;
+  val(value?: string): Cash;
```

Comment:
`value` here should be non-optional, since the first overload will cover that case anyway.